x-pack/plugins/lens/public/indexpattern_datasource/format_column.ts

```diff
@@ -117,13 +117,27 @@ export const formatColumn: ExpressionFunctionDefinition<
           }
           if (parentFormatParams) {
             const innerParams = (col.meta.params?.params as Record<string, unknown>) ?? {};
-            return withParams(col, {
-              ...col.meta.params,
-              params: {
-                ...innerParams,
-                ...parentFormatParams,
-              },
-            });
+            if (isNestedFormat(col.meta.params)) {
+              // original format is already a nested one, we are just replacing the wrapper params
+              return withParams(col, {
+                ...col.meta.params,
+                params: {
+                  ...innerParams,
+                  ...parentFormatParams,
+                },
+              });
+            } else {
+              // original format is not nested, wrapping it insifr parentFormatId/parentFormatParams
+              return withParams(col, {
+                ...col.meta.params,
+                id: parentFormatId,
+                params: {
+                  id: col.meta.params?.id,
+                  params: innerParams,
+                  ...parentFormatParams,
+                },
+              });
```

Comment:
I find this bit a bit hard to wrap my head around.
I would propose something alternative:

```js
const isNested = isNestedFormat(col.meta.params);
const innerParams = isNested ? {id: col.meta.params?.id, params: col.meta.params?.params } : col.meta.params?.params;

const formatId = isNested ? col.meta.params.id : parentFormatId;

return withParams(col, {
    ...col.meta.params,
    id: formatId,
    params: {
      ...innerParams,
      ...parentFormatParams
    }
});

```

Reply:
@dej611 I refactored the code like this, however there is a bug in there - the first ternary condition calculating the `innerParams` is flipped (it should be the nested params if the original formatter was nested, not the other way around)